src/core/pool_type.hpp

```diff
@@ -83,6 +84,9 @@ struct Pool : PoolBase {
 
 	static constexpr size_t MAX_SIZE = Tmax_size; ///< Make template parameter accessible from outside
 
+	using BitmapStorage = uint64_t;
```

Comment:
Maybe the following is a faster solution on builds with a 32 bits architecture. After all, if the compiler can't access the 64 bit registers, then it has to do this with two registers which might worse for performance than just splitting the calculations.
Note that I have not run any benchmarks, and that 32 bits builds are generally < 0.5% in the survey.
```suggestion
	using BitmapStorage = size_t;
```

Reply:
Seems fair, and I designed the rest of the code for this occurrence... clearly ;)

Reply:
More appropriate might be `uint_fast32_t`, which will typically be 64 bit on 64 bit native platforms, and 32 bit on those where 64 bit is not efficient. (https://stackoverflow.com/a/59593614)

Reply:
I looked into `uint_fast32_t`, but... MSVC x64's version is 32 bits and not 64 bits: https://godbolt.org/z/fY31o6v53
If that would've worked, I would've suggested that instead of `size_t`.